src/index.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @fileoverview This is an ES Module which exports `vizObserver`, see its function below.
+ *
+ * This uses a combination of IntersectionObserver and ResizeObserver to inform you when DOM
+ * nodes move around on a page (and also when they're resized). This might be useful if
+ * you're trying to track a complex HTML structure, perhaps one you don't control.
+ *
+ * It also works with target nodes inside Shadow DOM.
+ */
+
+/**
+ * This is an element's position relative to the page's documentElement.
+ */
+export interface Rect {
+  appear: boolean;
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+export interface Options {
+  signal?: AbortSignal;
+}
+
+// const debug = false;
+const root = document.documentElement;
+
+const handlers = new WeakMap<Element, () => void>();
+
+const activeObservers = new Set<() => void>();
+handlers.set(root, () => activeObservers.forEach((handler) => handler()));
+
+const ro = new ResizeObserver((entries) => {
```

Comment:
There should be memory and perf improvements if we use a single instance of a ResizeObserver instead of creating a new one each time `vizObserver()` is called. Using a weak map to store callbacks should prevent memory leaks.

At some point I might change it to `WeakMap<Element, (() => void)[]>` since it could be the case that multiple things might want to observer the same element (e.g. multiple arrows pointing to the same box)